### PR TITLE
Define initial contributing guidelines

### PR DESCRIPTION
**Pull request by Adda0**
This PR defines the initial contributing guidelines, primarily regarding the design style. Further improvements are work-in-progress and further TODOs. Feel free to improve the guidelines after this PR gets merged (we should therefore merge the PR quickly, so everyone can iterate on the `CONTRIBUTING.md` file).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Contributing guidelines
+
+If you'd like to contribute to the libmata,  please [fork the repository](https://github.com/VeriFIT/mata/fork),
+create a new feature branch, and finally [create a new pull request](https://github.com/VeriFIT/mata/compare).
+
+In case you run into some unexpected behaviour, error or anything suspicious either contact us directly through mail or
+[create a new issue](https://github.com/VeriFIT/mata/issues/new/choose).
+When creating a new issue, please, try to include everything necessary for us to know (such as the version, operation
```
Comment (jurajsic) [inline CONTRIBUTING.md]:
```suggestion
When creating a new issue, please, try to include everything necessary for us to know (such as the version, operating
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+system, etc.) so we can sucessfully replicate the issue.
+
+## Design style
+
+- private attributes of classes/structs should use `snake_case_` (note the trailing `_`).
```
Comment (jurajsic) [inline CONTRIBUTING.md]: Do we actually do that? Maybe put **private** in bold.

**Review by jurajsic — APPROVED** [review]
Looks good

File: CONTRIBUTING.md (hunk 2 above)
Comment (Adda0) [reply CONTRIBUTING.md]: As per our agreement on one of our previous Mata meetings:
1. we should for all new classes, and
2. we should rewrite all old classes to adhere to this convention one day, too.

I shall emphasize the word *private*.

**Pull request merged**